### PR TITLE
Bug 2117738: reset ErrorBoundary state on location change

### PR DESCRIPTION
**Pull request by rhamilto**
After:

https://user-images.githubusercontent.com/895728/184711465-e12535b5-cfe1-4b36-94da-e80a16d6fb19.mov

### CODE REVIEW

**Review by andrewballantyne — APPROVED** [review]

File: frontend/packages/console-shared/src/components/error/error-boundary.tsx [hunk 1]
```diff
@@ -15,19 +16,35 @@ export type ErrorBoundaryState = {
 const DefaultFallback: React.FC = () => <div />;
 
 class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  unlisten: () => void = () => {};
+
+  defaultState = {
```
[the remark this answers is not in the document]
Comment (rhamilto) [reply frontend/packages/console-shared/src/components/error/error-boundary.tsx]: Thanks, @andrewballantyne!  Updated.

File: frontend/packages/console-shared/src/components/error/error-boundary.tsx [hunk 2]
```diff
@@ -15,19 +16,35 @@ export type ErrorBoundaryState = {
...
-      },
-    };
+    this.state = this.defaultState;
+  }
+
+  componentDidMount() {
+    this.unlisten = history.listen(() => {
+      // reset state to default when location changes
+      this.setState(this.defaultState);
+      // }
```
Comment (spadgett) [inline frontend/packages/console-shared/src/components/error/error-boundary.tsx:42]: nit: accidental comment?

File: frontend/packages/console-shared/src/components/error/error-boundary.tsx [hunk 3]
```diff
@@ -15,19 +16,35 @@ export type ErrorBoundaryState = {
 const DefaultFallback: React.FC = () => <div />;
 
 class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  unlisten: () => void = () => {};
+
+  defaultState: ErrorBoundaryState = {
```
Comment (spadgett) [inline frontend/packages/console-shared/src/components/error/error-boundary.tsx:21]: Does `readonly` work here to make sure we don't accidentally mutate the value?

```suggestion
  readonly defaultState: ErrorBoundaryState = {
```

Comment (andrewballantyne) [reply frontend/packages/console-shared/src/components/error/error-boundary.tsx:21]: +1 -- TS will definitely get in the way of mutating it accidentally.

File: frontend/packages/console-shared/src/components/error/error-boundary.tsx (hunk 2 above)
Comment (rhamilto) [reply frontend/packages/console-shared/src/components/error/error-boundary.tsx]: Fixed.  Thanks.

File: frontend/packages/console-shared/src/components/error/error-boundary.tsx (hunk 3 above)
Comment (rhamilto) [reply frontend/packages/console-shared/src/components/error/error-boundary.tsx]: Added.

**Pull request merged**